data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,150 @@
+import me.androidbox.data.local.entity.TaskEntity
+import me.androidbox.data.mapper.toReminderDto
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.remote.network.reminder.ReminderService
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.constant.Constant.RETRY_COUNT
+import me.androidbox.domain.constant.SyncAgendaType
+
+@HiltWorker
+/** TODO Rename to something more readable as this is not just related to tasks */
+class TaskCreateSynchronizerWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val taskService: TaskService,
+    private val taskDao: TaskDao,
+    private val reminderService: ReminderService,
+    private val reminderDao: ReminderDao,
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+
+        /** TODO is this the correct place to have this check and to retry or fail */
+        if(runAttemptCount > RETRY_COUNT) {
+            Result.failure()
+        }
+        else {
+            Result.retry()
+        }
+
+        val createdTasks: Deferred<List<TaskEntity>>
+        val updatedTasks: Deferred<List<TaskEntity>>
+        val createdReminders: Deferred<List<ReminderEntity>>
+        val updatedReminders: Deferred<List<ReminderEntity>>
+
+        /** Get all the created, updated tasks and reminders
+         *  TODO As these will run sequentially - is it better to use supervisor scope and run them in parallel? */
+        supervisorScope {
+            createdTasks = async {
+                val taskIds = taskDao.getAllTasksBySyncType(SyncAgendaType.CREATE)
+
+                taskIds.map { id ->
+                    taskDao.getTaskById(id)
+                }
+            }
+
+            updatedTasks = async {
+                val taskIds = taskDao.getAllTasksBySyncType(SyncAgendaType.UPDATE)
+
+                taskIds.map { id ->
+                    taskDao.getTaskById(id)
+                }
+            }
+
+            createdReminders = async {
+                val reminderIds = reminderDao.getAllRemindersBySyncType(SyncAgendaType.CREATE)
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+
+            updatedReminders = async {
+                val reminderIds = reminderDao.getAllRemindersBySyncType(SyncAgendaType.UPDATE)
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+        }
+
+        /** TODO I am using supervisor scope to launch so if one fails the other will still run
+         *  However, will the worker return failure if a single one fails even though some failed?
+         *  My checkResult will run everything in this block */
```

Comment:
Yea it's fine to use a supervisorScope here

If even one fails, you should return a Result.retry()

Why? Because on the next run, since you removed some from the db, you will run only the remaining ones